haystack/preview/components/embedders/openai_text_embedder.py

```diff
@@ -0,0 +1,110 @@
+from typing import List, Optional, Dict, Any
+import os
+
+import openai
+from canals.errors import DeserializationError
+
+from haystack.preview import component, default_to_dict, default_from_dict
+
+
+API_BASE_URL = "https://api.openai.com/v1"
+
+
+@component
+class OpenAITextEmbedder:
+    """
+    A component for embedding strings using OpenAI models.
+    """
+
+    def __init__(
+        self,
+        api_key: Optional[str] = None,
+        model_name: str = "text-embedding-ada-002",
+        api_base_url: str = API_BASE_URL,
+        organization: Optional[str] = None,
+        prefix: str = "",
+        suffix: str = "",
+    ):
+        """
+        Create a OpenAITextEmbedder component.
+
+        :param api_key: The OpenAI API key.
+                        If not expressly provided, the API key will be read from the environment variable OPENAI_API_KEY.
+        :param model_name: The name of the model to use.
+        :param api_base_url: The OpenAI API Base url, defaults to `https://api.openai.com/v1`.
+        :param organization: The OpenAI-Organization ID, defaults to `None`. For more details, see OpenAI
+        [documentation](https://platform.openai.com/docs/api-reference/requesting-organization).
+        :param prefix: A string to add to the beginning of each text.
+        :param suffix: A string to add to the end of each text.
+        """
+
+        if api_key is None:
+            try:
+                api_key = os.environ["OPENAI_API_KEY"]
+            except KeyError:
+                raise ValueError(
+                    "OpenAITextEmbedder expects an OpenAI API key. "
+                    "Please provide one using the api_key parameter or set the environment variable OPENAI_API_KEY."
+                )
+
+        self.api_key = api_key
+        self.model_name = model_name
+        self.api_base_url = api_base_url
+        self.organization = organization
+        self.prefix = prefix
+        self.suffix = suffix
+
+    def to_dict(self) -> Dict[str, Any]:
+        """
+        Serialize this component to a dictionary.
+        """
+        return default_to_dict(
+            self,
+            api_key="OPENAI_API_KEY",
```

Comment:
I'd just skip it TBH. You serialize with no `api_key` field, when you de-serialize the constructor will get `api_key=None` and will go through the process of checking the env var.

Reply:
Please note that skipping the serialization part won't prevent the usage of secrets in yaml files. So it would still be possible to have `api_key: ${secrets.API_KEY}` in a `pipeline.yml` file that was manually created.

Reply:
Ok. This approach is simpler.
 I'll go ahead with it!